runtime/src/main/java/io/quarkiverse/freemarker/runtime/FreemarkerConfigurationProducer.java

```diff
@@ -30,17 +30,27 @@
 
     private static final Logger log = LoggerFactory.getLogger(FreemarkerConfigurationProducer.class);
 
+    private volatile List<String> resourcePaths;
+    private volatile Map<String, String> directives;
+
+    public void initiliaze(List<String> resourcePaths, Map<String, String> directives) {
+        this.resourcePaths = resourcePaths;
+        this.directives = directives;
+    }
+
     @DefaultBean
     @Singleton
     @Produces
-    public Configuration configuration(FreemarkerBuildConfig buildConfig, FreemarkerConfig config)
+    public Configuration configuration(FreemarkerConfig config)
             throws ClassNotFoundException, IllegalAccessException, InstantiationException {
 
         Configuration cfg = new Configuration(Configuration.VERSION_2_3_30);
 
         List<TemplateLoader> loaders = new ArrayList<>();
-        log.info("Adding build time locations " + buildConfig.resourcePaths);
-        loaders.addAll(buildConfig.resourcePaths.stream().map(this::newClassTemplateLoader).collect(toList()));
+        log.info("Adding build time locations " + resourcePaths);
+        resourcePaths.stream()
+                .map(this::newClassTemplateLoader)
+                .forEach(loaders::add);
 
         log.info("Adding runtime locations " + config.filePaths.orElse(emptyList()));
```

Comment:
What benefits would it bring? For external locations, we need just the base paths to be able to produce the configuration. We do not need to know the set of the template files because we do not want to include them into the native image. I think it would just make the config more complex for the end users without any technical reason that would require that. But maybe I am overseeing something?

Reply:
the idea was configuration consistency. if your external templates was a folder containing far more than what you really needed fo your app, you could start including and excluding things, to prevent from fetching the wrong template at runtime.
but I checked the `FileTemplateLoader`, and it does not provide that sort of flexibility.

Reply:
Yes, exactly, as long as Freemarker's own FileTemplateLoader does not support includes/excludes, there is no point in having them in our config.